### PR TITLE
Replace `GetMaxRoundLoaded()` with `GetNextRoundToLoad()`

### PR DESCRIPTION
**Pull request by tolikzinovyev**
Also fixes a bug in `GetMaxRoundLoaded()`.

### CODE REVIEW

File: idb/postgres/postgres.go [hunk 1]
```diff
@@ -399,24 +399,23 @@ func (db *IndexerDb) GetMaxRoundAccounted() (round uint64, err error) {
...
-		return
+		return 0, err
 	}
 
-	round = uint64(nullableRound.Int64)
-	return
+	if !nullableRound.Valid {
+		return 0, nil
+	}
+	return uint64(nullableRound.Int64), nil
```
Comment (winder) [inline idb/postgres/postgres.go:418]:
```suggestion
	return uint64(nullableRound.Int64) + 1, nil
```

File: idb/postgres/postgres.go [hunk 2]
```diff
@@ -399,24 +399,23 @@ func (db *IndexerDb) GetMaxRoundAccounted() (round uint64, err error) {
...
 	row := db.db.QueryRow(`SELECT max(round) FROM block_header`)
-	err = row.Scan(&nullableRound)
 
-	if err == sql.ErrNoRows || !nullableRound.Valid {
-		err = idb.ErrorNotInitialized
-		return
+	var nullableRound sql.NullInt64
+	err := row.Scan(&nullableRound)
+	if err == sql.ErrNoRows {
+		return 0, nil
```
Comment (winder) [inline idb/postgres/postgres.go:409]: Return  `idb.ErrorNotInitialized`?

File: idb/postgres/postgres_integration_test.go [hunk 3]
```diff
@@ -44,14 +44,15 @@ func TestMaxRoundOnUninitializedDB(t *testing.T) {
...
+	roundL, errL := db.GetNextRoundToLoad()
 
 	//////////
 	// Then // The error message should be set.
 	//////////
 	assert.Equal(t, errA, idb.ErrorNotInitialized)
-	assert.Equal(t, errL, idb.ErrorNotInitialized)
 	assert.Equal(t, uint64(0), roundA)
+
+	require.NoError(t, errL)
```
Comment (winder) [inline idb/postgres/postgres_integration_test.go:55]: Why is this changing?

File: idb/postgres/postgres_integration_test.go [hunk 4]
```diff
@@ -95,14 +96,14 @@ func TestMaxRound(t *testing.T) {
...
-	roundL, err := pdb.GetMaxRoundLoaded()
+	roundL, err := pdb.GetNextRoundToLoad()
 	assert.NoError(t, err)
 
 	//////////
 	// Then // There should be no error and we return that there are zero rounds.
 	//////////
 	assert.Equal(t, uint64(123454321), roundA)
-	assert.Equal(t, uint64(543212345), roundL)
+	assert.Equal(t, uint64(543212346), roundL)
```
Comment (winder) [inline idb/postgres/postgres_integration_test.go:106]: I'm confused about how this test is passing, where does the +1 happen?

**Review by winder — CHANGES_REQUESTED** [review]
I'm not sure about these changes, seems like there could be a regression.

File: idb/postgres/postgres.go (hunk 2 above)
Comment (tolikzinovyev) [reply idb/postgres/postgres.go:409]: I think it shouldn't be a thing for importing blocks. With accounting it's genesis that needs to be initialized, here it's nothing.

File: idb/postgres/postgres.go (hunk 1 above)
Comment (tolikzinovyev) [reply idb/postgres/postgres.go]: right, thanks

File: idb/postgres/postgres_integration_test.go (hunk 4 above)
Comment (tolikzinovyev) [reply idb/postgres/postgres_integration_test.go:106]: It was not. After fixing the bug you pointed out, it is.

**Review by winder — APPROVED** [review]

**Pull request merged**